src/service/internal_tables_access.h

```diff
@@ -331,10 +333,35 @@ namespace ccf
       if (service->has())
       {
         const auto prev_service_info = service->get();
+
         auto previous_service_identity = tx.wo<ccf::PreviousServiceIdentity>(
           ccf::Tables::PREVIOUS_SERVICE_IDENTITY);
         previous_service_identity->put(prev_service_info->cert);
 
+        auto previous_identity_endorsement =
+          tx.wo<ccf::PreviousServiceIdentityEndorsement>(
+            ccf::Tables::PREVIOUS_SERVICE_IDENTITY_ENDORSEMENT);
+        try
+        {
+          const auto pubkey = ccf::crypto::public_key_pem_from_cert(
```

Comment:
Talked offline, agreed we'll move it to the opening TX to avoid doing things during recovery which are not really needed for recovery, and producing back endorsements seems like one of them.